tests/BenchmarkDotNet.Tests/Validators/ParamsValidatorTests.cs

```diff
@@ -0,0 +1,206 @@
+        }
+
+        public class Const1 : Base
+        {
+            [Params(false, true)]
+            public const bool Input = false;
+        }
+
+        public class Const2 : Base
+        {
+            [ParamsAllValues]
+            public const bool Input = false;
+        }
+
+        public class Const3 : Base
+        {
+            [ParamsSource(nameof(Source))]
+            public const bool Input = false;
+        }
+
+        public class StaticReadonly1 : Base
+        {
+            [Params(false, true)]
+            public static readonly bool Input = false;
+        }
+
+        public class StaticReadonly2 : Base
+        {
+            [ParamsAllValues]
+            public static readonly bool Input = false;
+        }
+
+        public class StaticReadonly3 : Base
+        {
+            [ParamsSource(nameof(Source))]
+            public static readonly bool Input = false;
+        }
+
+        public class NonStaticReadonly1 : Base
+        {
+            [Params(false, true)]
+            public readonly bool Input = false;
+        }
+
+        public class NonStaticReadonly2 : Base
+        {
+            [ParamsAllValues]
+            public readonly bool Input = false;
+        }
+
+        public class NonStaticReadonly3 : Base
+        {
+            [ParamsSource(nameof(Source))]
+            public readonly bool Input = false;
+        }
+
+        public class FieldMultiple1 : Base
+        {
+            [Params(false, true)]
+            [ParamsAllValues]
+            public bool Input = false;
+        }
+
+        public class FieldMultiple2 : Base
+        {
+            [Params(false, true)]
+            [ParamsSource(nameof(Source))]
+            public bool Input = false;
+        }
+
+        public class FieldMultiple3 : Base
+        {
+            [ParamsAllValues]
+            [ParamsSource(nameof(Source))]
+            public bool Input = false;
+        }
+
+        public class FieldMultiple4 : Base
+        {
+            [Params(false, true)]
+            [ParamsAllValues]
+            [ParamsSource(nameof(Source))]
+            public bool Input = false;
+        }
+
+        public class PropMultiple1 : Base
+        {
+            [Params(false, true)]
+            [ParamsAllValues]
+            public bool Input { get; set; }
+        }
+
+        public class PropMultiple2 : Base
+        {
+            [Params(false, true)]
+            [ParamsSource(nameof(Source))]
+            public bool Input { get; set; }
+        }
+
+        public class PropMultiple3 : Base
+        {
+            [ParamsAllValues]
+            [ParamsSource(nameof(Source))]
+            public bool Input { get; set; }
+        }
+
+        public class PropMultiple4 : Base
+        {
+            [Params(false, true)]
+            [ParamsAllValues]
+            [ParamsSource(nameof(Source))]
+            public bool Input { get; set; }
+        }
+
```

Comment:
Add tests for `private set` and no setter?

Reply:
It's tricky. Currently, if no public setter is available, BenchmarkDotNet throws [this exception](https://github.com/dotnet/BenchmarkDotNet/blob/v0.13.5/src/BenchmarkDotNet/Extensions/ReflectionExtensions.cs#L181):
```cs
Unhandled exception. System.InvalidOperationException: Member "Input" must be public if it has the [ParamsAttribute] attribute applied to it
   at BenchmarkDotNet.Extensions.ReflectionExtensions.GetTypeMembersWithGivenAttribute[TAttribute](Type type, BindingFlags reflectionFlags)
   at BenchmarkDotNet.Running.BenchmarkConverter.<>c__DisplayClass11_0.<GetParameterDefinitions>g__GetDefinitions|0[TAttribute](Func`3 getValidValues)
   at BenchmarkDotNet.Running.BenchmarkConverter.GetParameterDefinitions(Type type)
   at BenchmarkDotNet.Running.BenchmarkConverter.MethodsToBenchmarksWithFullConfig(Type type, MethodInfo[] benchmarkMethods, IConfig config)
   at BenchmarkDotNet.Running.BenchmarkConverter.TypeToBenchmarks(Type type, IConfig config)
   at BenchmarkDotNet.Running.BenchmarkRunner.RunWithDirtyAssemblyResolveHelper(Type type, IConfig config, String[] args)
   at BenchmarkDotNet.Running.BenchmarkRunner.<>c__DisplayClass0_0`1.<Run>b__0()
   at BenchmarkDotNet.Running.BenchmarkRunner.RunWithExceptionHandling(Func`1 run)
   at BenchmarkDotNet.Running.BenchmarkRunner.Run[T](IConfig config, String[] args)
```
This happens before the validation stage.

I think it would be better to abandon such exceptions in favor of `Validator` checks (this exception was introduced in 2017). However, it deserves a separate PR (I can work on it once this PR is merged).

Reply:
@AndreyAkinshin That's the case for `public bool Input { get; private set; }`? The code looks like it just checks if the property itself is public, not its setter.